flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/RetryValidationStrategy.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.base.sink.writer;
+
+import org.apache.flink.annotation.Internal;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/** Class acting as a classifier for exception by failed requests to be retried by sink. */
+@Internal
+public class RetryValidationStrategy {
```

Comment:
The word `Validation` in this class is confusing. Seems like it is a strategy determining when to retry validation. Suggest we make it more general `RetryStrategy` or maybe `ExceptionClassifier` but make sure there are no clashes with existing classes. This class is in the `sink` package but it looks more general than sink

Reply:
I tried going with `RetryStrategy` but this clashes with another interface in `flink-core`. `ExceptionClassifier` seems suitable.
I also agree with having it in a more general module, `flink-core` I believe should be suitable.